azure/hub/variables.tf

```diff
@@ -232,7 +232,7 @@ variable "network_watcher_config" {
   description = "Configuration for the network watcher resource."
   type = object({
     name                = optional(string)
```

Comment:
Would there ever be a situation where name should be optional. IE would a user ever want to only set the resource group but not the name? Would it make more sense to set name to `string` instead of `optional(string)` whilst keeping resource group optional.

Just a thought, I'm open to suggestion here.